chainercv/datasets/mixup_dataset.py

```diff
@@ -0,0 +1,60 @@
+import numpy
+
+from chainer.dataset import dataset_mixin
+
+
+class MixUpSoftLabelDataset(dataset_mixin.DatasetMixin):
+
+    """Dataset which returns mixed images and labels for mixup learning[1].
+
+    `MixUpSoftLabelDataset` mix two images and labels which is chosen by base
+    dataset respectively by weighted average.
+
+    Unlike `LabeledImageDatasets`, label is a one-dimensional float array with
+    at most two nonzero weights (i.e. soft label). The summed weights is one.
+
+    The base dataset `__getitem__` should return image and label. Please see
+    the following example.
+
+    Example:
+
+        We construct a mixup dataset from MNIST.
+
+        .. code::
+
+            >>> from chainer.datasets import get_mnist
+            >>> from chainercv.datasets import SiameseDataset
+            >>> from chainercv.datasets import MixUpSoftLabelDataset
+            >>> mnist, _ = get_mnist()
+            >>> base_dataset = SiameseDataset(mnist, mnist)
+            >>> dataset = MixUpSoftLabelDataset(base_dataset)
+            >>> mixed_image, mixed_label = dataset[0]
+
+    Args:
+        dataset: The underlying dataset. dataset should returns two image
+            and their label. Typically, dataset is `SiameseDataset`.
+            More over, each element of each dataset should have same shape.
+            dataset also needs to support `__len__` as described above.
+        max_label: Max label index of base datasets.
+
+    See the papers for details: [1]`mixup: Beyond Empirical Risk Minimization
+    <https://arxiv.org/abs/1710.09412>`_.
+
+    """
+
+    def __init__(self, dataset, max_label):
```

Comment:
How about using `n_class` instead of `max_label`?
`n_class` is widely used in other parts of ChainerCV.

In that case the doc would be

```
n_class (int): The number of classes in the base dataset.
```

Reply:
You can remove `+1 in line 57 if you do this.